src/app/adapters/fallbackAppAdapter.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+
+import {fallbackDeclaration} from "../../analysis/fallbackDeclaration";
+import type {
+  StaticAppAdapter,
+  DynamicAppAdapter,
+} from "../../app/adapters/appAdapter";
+import {Assets} from "../../app/assets";
+import {type RepoId} from "../../core/repoId";
+import {Graph, NodeAddress, type NodeAddressT} from "../../core/graph";
+
+export class FallbackStaticAdapter implements StaticAppAdapter {
+  declaration() {
+    return fallbackDeclaration;
+  }
+
+  load(_unused_assets: Assets, _unused_repoId: RepoId) {
+    return Promise.resolve(new FallbackDynamicAdapter());
+  }
+}
+
+export class FallbackDynamicAdapter implements DynamicAppAdapter {
+  graph() {
+    return new Graph();
+  }
+
+  nodeDescription(x: NodeAddressT) {
+    return `[fallback]: ${NodeAddress.toString(x)}`;
```

Comment:
We probably don't want the `[fallback]: ` here, right? as this can appear in production?

Reply:
Fallback *shouldn't* appear in production. If it does, having the string [fallback] making it clear that the fallback plugin is present is probably a feature. It makes it easy to grep for the string and find why it appears.

Reply:
Okay, I'm persuaded.